mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -138,13 +143,13 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
       val hashDistCol = hashDistUDF(col($(outputCol)))
 
       // Compute threshold to get exact k elements.
-      // TODO: SPARK-18409: Use approxQuantile to get the threshold
-      val modelDatasetSortedByHash = modelDataset.sort(hashDistCol).limit(numNearestNeighbors)
-      val thresholdDataset = modelDatasetSortedByHash.select(max(hashDistCol))
-      val hashThreshold = thresholdDataset.take(1).head.getDouble(0)
+      val quantile = numNearestNeighbors.toDouble / modelDataset.count()
+      val modelDatasetWithDist = modelDataset.withColumn(distCol, hashDistUDF(col($(outputCol))))
```

Comment:
`hashDistCol` was defined earlier as `hashDistUDF(col($(outputCol)))`. We can use it.

Reply:
Thanks for your comment. I will update the code.